c/httpfileservice.c

```diff
@@ -452,6 +460,76 @@ void respondWithUnixFileMetadata(HttpResponse *response, char *absolutePath) {
   }
 }
 
+void directoryChangeOwnerAndRespond(HttpResponse *response, char *path,
+        char *usrId, char *grpId, char *Recursive, char *pattern) {
```

Comment:
Can you please show an example of a pattern and how it's used with this function?

Reply:
You can specify something like ".txt" which will select any file with this string in it, not necessarily the ending file name.  Or something like "foo" which will include any directory/file which contains the sub-string "foo".  It will not work with wildcards/

Reply:
Is this something that's been discussed and people agreed on such behavior?

Reply:
This has not been wildly discussed.  It has been seen in other code reviews.  In each instance, the pattern check has been localized to 1 function.  I figured that we may need to modify this check sometime in the future.